libmscore/edit.cpp

```diff
@@ -1534,6 +1535,119 @@ void Score::regroupNotesAndRests(int startTick, int endTick, int track)
+                              Tie* t = n->tieBack();
+                              Note* tnb = t ? t->startNote() : 0;
+                              if (tnb && tnb->chord()->measure() == chord->measure())
+                                    continue; // already accounted for this note
+                              int ticks = chord->duration().ticks();
+                              Note* tnf = 0;
+                              Note* nn = n;
+                              for (t = n->tieFor(); t; t = nn->tieFor()) {
+                                    nn = t->endNote();
+                                    if (nn->tick() + nn->chord()->duration().ticks() <= maxTick) {
+                                          ticks += nn->chord()->duration().ticks();
+                                          }
+                                    else {
+                                          tnf = nn;
+                                          break;
+                                          }
+                                    }
+                              TimeSigFrac nominal = sigmap()->timesig(seg->tick()).nominal();
+                              std::vector<TDuration> dList = toRhythmicDurationList(
+                                    Fraction::fromTicks(ticks), false, n->rtick(), nominal, seg->measure(), 1);
+                              vm.addTiedNotes(n->pitch(), n->tick(), dList, tnf, tnb); // add notes to temporary container
+                              }
+                        }
+                  }
+            // calculate the new voicing
+            vm.arrange();
+            // overwrite the real measure with the newly calculated one
+            for (int voice = 0, track = staffIdx * VOICES; voice < VOICES; voice++, track++) {
+                  doLayoutRange(startSeg->tick(), maxTick);
+                  Segment* seg = firstSeg;
+                  if (voice >= vm.numVoices()) {
+                        // no more vvoices so overwrite remaining voices with rests
+                        seg = setNoteRest(seg, track, NoteVal(), Fraction::fromTicks(maxTick - seg->tick()), Direction::AUTO);
+                        continue;
+                        }
+                  VirtualVoice* vvoice = vm.voices()[voice];
+                  for (RawChord* rchord : vvoice->chords()) {
+                        if (seg->tick() < rchord->tick()) {
+                              seg = setNoteRest(seg, track, NoteVal(), Fraction::fromTicks(rchord->tick() - seg->tick()), Direction::AUTO); // This *appears* to work in all voices, but see FIXME below.
+                              while (seg->tick() < rchord->tick()) // might encounter segments from other voices before the one we want
+                                    seg = seg->next(Segment::Type::ChordRest); // FIXME: in higher voices this sometimes does not find the ChordRest we just added 2 lines above with setNoteRest()
+                              }
```

Comment:
This is where the problem occurs.
I use [`setNoteRest`](https://github.com/musescore/MuseScore/pull/2794/files#diff-1b7205502c863c4426ee75767ddb1ea4R1599) to add a rest so that I can guarantee there will be a ChordRest at the tick immediately after it. I then try to move to the end of the rest with [`seg->next(Segment::Type::ChordRest)`](https://github.com/musescore/MuseScore/pull/2794/files#diff-1b7205502c863c4426ee75767ddb1ea4R1601). This works fine in Voice 1 and usually in higher voices too, but sometimes in higher voices it moves too far and skips past the end of the rest.